app/models/piece.rb

```diff
@@ -73,4 +74,24 @@ def move_to!(new_x, new_y)
       self.update(x_position: new_x, y_position: new_y)
     end
   end
+=======
+  def moving_on_board?(x, y)
+    x.between?(0, 7) && y.between?(0, 7)
+  end
```

Comment:
I think this just checks if x and y are on the board? I don't know if this is necessary since only values of 0 to 7 can be used in the update.  The square id's are where x and y are pulled from and they can only be between 0 and 7.

Reply:
I'm thinking that may be a silly unneeded method looking at it now.